gsitemap.php

```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
         return true;
     }
 
+    /**
+	 * return the link elements for the manufacturer object
+	 *
+	 * @param array  $link_sitemap    contain all the links for the Google Sitemap file to be generated
+	 * @param string $lang            language of link to add
+	 * @param int    $index           index of the current Google Sitemap file
+	 * @param int    $i               count of elements added to sitemap main array
+	 * @param int    $id_manufacturer manufacturer object identifier
+	 *
+	 * @return bool
+	 */
+	protected function getManufacturerLink(&$link_sitemap, $lang, &$index, &$i, $id_manufacturer = 0)
+	{
+		$link = new Link();
+		if (method_exists('ShopUrl', 'resetMainDomainCache'))
+			ShopUrl::resetMainDomainCache();
+		$manufacturers_id = Db::getInstance()->ExecuteS(
+			'SELECT m.`id_manufacturer` FROM `'._DB_PREFIX_.'manufacturer` m
+			INNER JOIN `'._DB_PREFIX_.'manufacturer_lang` ml on m.`id_manufacturer` = ml.`id_manufacturer`'.
+			($this->tableColumnExists(_DB_PREFIX_.'manufacturer_shop') ? ' INNER JOIN `'._DB_PREFIX_.'manufacturer_shop` ms ON m.`id_manufacturer` = ms.`id_manufacturer` ' : '').
+			' WHERE m.`active` = 1  AND m.`id_manufacturer` >= '.(int)$id_manufacturer.
+			($this->tableColumnExists(_DB_PREFIX_.'manufacturer_shop') ? ' AND ms.`id_shop` = '.(int)$this->context->shop->id : '').
+			' AND ml.`id_lang` = '.(int)$lang['id_lang'].
+			' ORDER BY m.`id_manufacturer` ASC'
+		);
+		foreach ($manufacturers_id as $manufacturer_id)
+		{
+			$manufacturer = new Manufacturer((int)$manufacturer_id['id_manufacturer'], $lang['id_lang']);
+			$url = $link->getManufacturerLink($manufacturer, $manufacturer->link_rewrite, $lang['id_lang']);
+
+			$image_link = 'http'.(Configuration::get('PS_SSL_ENABLED') && Configuration::get('PS_SSL_ENABLED_EVERYWHERE') ? 's' : '').'://'.Tools::getMediaServer(_THEME_MANU_DIR_)._THEME_MANU_DIR_.((!file_exists(_PS_MANU_IMG_DIR_.'/'.(int)$manufacturer->id.'-medium_default.jpg')) ? $lang['iso_code'].'-default' : (int)$manufacturer->id).'-medium_default.jpg';
+			$image_link = (!in_array(rtrim(Context::getContext()->shop->virtual_uri, '/'), explode('/', $image_link))) ? str_replace(
```

Comment:
This code needs to be refactor, it's the same as above, and doesn't have comments.